controllers/helpers.go

```diff
@@ -182,10 +183,11 @@ func getPEUUID(client *nutanixClientV3.Client, peName, peUUID *string) (string,
 		} else {
 			foundPEUUID = *foundPEs[0].Metadata.UUID
 		}
-		if foundPEUUID == "" {
-			return "", fmt.Errorf("failed to retrieve Prism Element cluster by name or uuid. Verify input parameters.")
-		}
 	}
+	if foundPEUUID == "" {
+		return "", fmt.Errorf("failed to retrieve Prism Element cluster by name or uuid. Verify input parameters.")
+	}
+
```

Comment:
Adding a comment here because it won't let me at the function declaration. My understanding is that the function is supposed to return the PE UUID given a name or a UUID. This smells; why is the UUID a parameter?

Reply:
something like this is a lot cleaner
```go
func getPrismelement(client *nutanixClientV3.Client, peName, peUUID string) (string, error) {
	if peUUID != "" {
	    // UUID already exists just verify if it corresponds to a real PE
	    peIntentResponse, err := client.V3.GetCluster(peUUID)
		if err != nil {
			if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
				return "", fmt.Errorf("failed to find Prism Element cluster with UUID %s: %v", *peUUID, err)
			}
		}
		return *peIntentResponse.Metadata.UUID, nil
	}
	if peName != "" {
		filter := getFilterForName(peName)
		responsePEs, err := client.V3.ListAllCluster(filter)
		if err != nil {
			return "", err
		}
		// Validate filtered PEs
		foundPEs := make([]*nutanixClientV3.ClusterIntentResponse, 0)
		for _, s := range responsePEs.Entities {
			peSpec := s.Spec
			if *peSpec.Name == peName && hasPEClusterServiceEnabled(s, serviceNamePECluster) {
				foundPEs = append(foundPEs, s)
			}
		}
		if len(foundPEs) == 1 {
			return *foundPEs[0].Metadata.UUID
		} else if len(foundPEs) == 0 {
			return "", fmt.Errorf("failed to retrieve Prism Element cluster by name %s", peName)
		} else {
			return "", fmt.Errorf("more than one Prism Element cluster found with name %s", peName)
		}
	}

	return "", fmt.Errorf("failed to retrieve Prism Element cluster by name %s or uuid %s", peName, peUUID)
}
```